src/spacyMatcher.py

```diff
@@ -1,54 +1,49 @@
-##        If TAG = '' will return all TAGs 
-## Output: List of dictionaries for all TAGs found in the text (os a specific tag if passed)
-## Author: The Sentimentalists / Ana B Potje
-## Date: 15/Oct/2020
-##
-## This function uses the Python Library Spacy. The user will input an URL and a "TAG" such as:
-##    PERSON - People, including fictional.
-##    ORG - Companies, agencies, institutions, etc.
-##    GPE - Countries, cities, states.
-##    PERCENT - Percentage, including ”%“.
-##    LANGUAGE - Any named language.
-##    DATE - Absolute or relative dates or periods.
-##    TIME - Times smaller than a day.
-##    LOC - Non-GPE locations, mountain ranges, bodies of water.
-##    NORP - Nationalities or religious or political groups.
-##    EVENT - Named hurricanes, battles, wars, sports events, etc.
-##    WORK_OF_ART - Titles of books, songs, etc.
-##    MONEY - Monetary values, including unit.
-##    QUANTITY - Measurements, as of weight or distance.
-##    ORDINAL - “first”, “second”, etc.
-##    CARDINAL - Numerals that do not fall under another type (not ordinal, quantity ..)
-##
-## It will return a list with distinct values of all occurrences of that entity in the text.
-###########################################################################################################
-
+"""
+This module contains a a function that returns named entities from text
+"""
 from aws_xray_sdk.core import xray_recorder
+import spacy
 
 @xray_recorder.capture('spacyMatcher')
-def spacyMatcher(text, tag): 
-    import spacy
-
-    if tag == '':
-        all_tags = ['PERSON','GPE','ORG','PERCENT','LANGUAGE','DATE','TIME','LOC','NORP','EVENT','WORK_OF_ART',
+def spacyMatcher(text, tag=None):
+    """
+    This function takes text and an optional tag.
+    It returns a dictionary tags and the entities corresponding to that tag.
+    
+    If no tag is specified, all tags will be used.
+    
+    Tags are SpaCy tags:
+    PERSON - People, including fictional.
+    ORG - Companies, agencies, institutions, etc.
+    GPE - Countries, cities, states.
+    PERCENT - Percentage, including ”%“.
+    LANGUAGE - Any named language.
+    DATE - Absolute or relative dates or periods.
+    TIME - Times smaller than a day.
+    LOC - Non-GPE locations, mountain ranges, bodies of water.
+    NORP - Nationalities or religious or political groups.
+    EVENT - Named hurricanes, battles, wars, sports events, etc.
+    WORK_OF_ART - Titles of books, songs, etc.
+    MONEY - Monetary values, including unit.
+    QUANTITY - Measurements, as of weight or distance.
+    ORDINAL - “first”, “second”, etc.
+    CARDINAL - Numerals that do not fall under another type (not ordinal, quantity ..).
+    """
+     all_tags = ['PERSON','GPE','ORG','PERCENT','LANGUAGE','DATE','TIME','LOC','NORP','EVENT','WORK_OF_ART',
                     'MONEY','QUANTITY','ORDINAL','CARDINAL']
-    else:
-        all_tags = [tag]
+        
+    tags = [tag] if tag else all_tags
```

Comment:
I think this is neater than the original big if statement than you had and it seems to be the way people prefer to handle "defaults" in python

Reply:
Great tip ;)

Reply:
Also, if you changed it so people needed to pass a list of tags rather than a single tag, you could make this even simpler because you could just put: `tags = tag or all_tags` this is a shorthand for `tags= tag if tag else all_tags` but it will only work if tag is a list because otherwise you need to convert it to a list.

Reply:
The "spacy matcher" function was our last one to implement, it was completed on Friday (30th Oct) after the 4th Sprint review, then frontend "React" tags were created during the weekend (days before the final presentation!).
Therefore looking at it now we can clearly see improvements. The "spacy matcher" had an original purpose of being used for one tag - or all tags. But it is in fact being used only for all tags! So there is no need for a TAG parameter after all ...
Will fix this logic too - many thanks again!